lib/Firtool/Firtool.cpp

```diff
@@ -228,11 +228,14 @@ LogicalResult firtool::populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
   //
   // TODO: Improve LowerLayers to avoid the need for canonicalization. See:
   //   https://github.com/llvm/circt/issues/7896
-  if (opt.shouldAdvancedLayerSink())
-    pm.nest<firrtl::CircuitOp>().addPass(firrtl::createAdvancedLayerSinkPass());
-  else
-    pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
-        firrtl::createLayerSinkPass());
+  if (opt.shouldDisableLayerSink()) {
+    if (opt.shouldAdvancedLayerSink())
+      pm.nest<firrtl::CircuitOp>().addPass(
+          firrtl::createAdvancedLayerSinkPass());
+    else
+      pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
+          firrtl::createLayerSinkPass());
+  }
```

Comment:
Given that we now have two options, it doesn't seem great that ALS is gated by another option. I.e., `-advanced-layer-sink` has no effect if `-disable-layer-sink` is set.

Would it be better to instead just make two simple options:

1. `-advanced-layer-sink` : schedules advanced layer sink
2. `-layer-sink` : schedules layer sink

Alternatively an enum that has the three actually interesting states: `-layer-sink=none,simple,advanced`.

Reply:
I figured since we're going to fully replace LS with ALS and removing `--advanced-layer-sink`, that this would have the least churn. Then, we could keep the `disable-layer-sink` flag as long as we find it useful.  Moving to an enum would have to be reversed when we remove either flag.  I'm not sure about the first idea, I guess we would have to disable layersink by default to make it work?

Reply:
Yeah, it would have to be default of LS on and ALS off if there are two options.